include/unifex/task.hpp

```diff
@@ -226,6 +227,21 @@ struct _awaiter {
   };
 };
 
+struct _stop_if_requested {
+  bool await_ready() const noexcept {
+    return false;
+  }
+  template <typename Promise>
+  coro::coroutine_handle<> await_suspend(coro::coroutine_handle<Promise> coro) const noexcept {
+    if (get_stop_token(coro.promise()).stop_requested()) {
+      return coro.promise().unhandled_done();
+    }
+    return coro; // don't suspend
+  }
+  void await_resume() const noexcept {
+  }
+};
+
```

Comment:
Could this algorithm be implemented as a sender that could then be used within other composed operations rather than just being usable by coroutines?

Reply:
Possibly! I'll play with it.

Reply:
It occurs to me that if it's implemented as a sender algorithm, we don't get to benefit from symmetric transfer. I'm not sure whether that matters here or not.